pkg/reward/reward.go

```diff
@@ -37,33 +44,34 @@ type ContractCaller interface {
 	}, error)
 }
 
-func New(logger log.Logger, aggr aggregator.IAggregator, contractCaller ContractCaller, client contracts.ETHClient, account *eth.Account) *Reward {
-	// Getting abi.
-	abi, _ := abi.JSON(strings.NewReader(tellor.TellorABI))
+func New(logger log.Logger, aggr aggregator.IAggregator, contractCaller ContractCaller) *Reward {
 	return &Reward{
 		aggr:           aggr,
 		logger:         log.With(logger, "component", ComponentName),
 		contractCaller: contractCaller,
 		gasUsed:        make(map[int64]*big.Int),
-		client:         client,
-		abi:            abi,
-		account:        account,
+		estimatedGasCostValue: promauto.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
we also need to record the actual gas used so we can compare them. You can get it from the TX receipt.

Reply:
I added it to the submitter. the estimation happens before the actual submit, so maybe we need to record its value somewhere in the submitter. IDK how it's possible to add it to the reward code?

Reply:
yes add it in the submitter, there we have the estimation and then when the tx is confirmed we have the actual cost